lib/msal-common/src/client/AuthorizationCodeClient.ts

```diff
@@ -45,13 +52,38 @@ export class AuthorizationCodeClient extends BaseClient {
      * API to acquire a token in exchange of 'authorization_code` acquired by the user in the first leg of the authorization_code_grant
      * @param request
      */
-    async acquireToken(request: AuthorizationCodeRequest): Promise<string> {
+    async acquireToken(request: AuthorizationCodeRequest): Promise<AuthenticationResult> {
 
         this.logger.info("in acquireToken call");
+
         const authority: Authority = await this.createAuthority(request && request.authority);
         const response = await this.executeTokenRequest(authority, request);
-        return JSON.stringify(response.body);
-        // TODO add response_handler here to send the response
+
+        const responseHandler = new ResponseHandler(
+            this.clientConfig.authOptions.clientId,
+            this.unifiedCacheManager,
+            this.cryptoUtils,
+            this.logger
+        );
+
+        responseHandler.validateServerAuthorizationTokenResponse(response.body);
+        const tokenResponse = await responseHandler.generateAuthenticationResult(
+            response.body,
+            authority
+        );
+
+        // set the final cache and return the auth response
+        this.setCache();
+        return tokenResponse;
+    }
+
+     /**
+     * Set the cache post acquireToken call
+     */
+    private setCache() {
+        const inMemCache = this.unifiedCacheManager.getCacheInMemory();
+        const cache = this.unifiedCacheManager.generateJsonCache(inMemCache);
+        this.cacheStorage.setSerializedCache(Serializer.serializeJSONBlob(cache));
```

Comment:
I think all of this should be in BaseClient, as it will be shared among all clients.

Reply:
Done